Sources/SwiftSyntax/SyntaxFactory.swift.gyb

```diff
@@ -199,10 +199,15 @@ public enum SyntaxFactory {
   public static func makeStringLiteralExpr(_ text: String,
     leadingTrivia: Trivia = [],
     trailingTrivia: Trivia = []) -> StringLiteralExprSyntax {
-    let literal = makeStringLiteral("\"\(text)\"", 
-                                    leadingTrivia: leadingTrivia,
-                                    trailingTrivia: trailingTrivia)
-    return makeStringLiteralExpr(stringLiteral: literal)
+    let string = makeStringSegment(text,
+                                   leadingTrivia: leadingTrivia,
+                                   trailingTrivia: trailingTrivia)
+    let segment = makeStringSegment(content: string)
+    let segments = makeStringLiteralSegments([segment])
+    let quote = makeStringQuoteToken()
+    return makeStringLiteralExpr(openQuote: quote,
+                                 segments: segments,
+                                 closeQuote: quote)
```

Comment:
I believe `leadingTrivia` and `trailingTrivia` should be attached to `openQuote` and `closeQuote`.

Reply:
addressed your feedback